pkg/controller/plan/adapter/ova/builder.go

```diff
@@ -0,0 +1,594 @@
+	core "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	cnv "kubevirt.io/api/core/v1"
+	cdi "kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1"
+)
+
+// BIOS types
+const (
+	Efi = "efi"
+)
+
+// Bus types
+const (
+	Virtio = "virtio"
+)
+
+// Input types
+const (
+	Tablet = "tablet"
+)
+
+// Network types
+const (
+	Pod    = "pod"
+	Multus = "multus"
+)
+
+// Template labels
+const (
+	TemplateOSLabel       = "os.template.kubevirt.io/%s"
+	TemplateWorkloadLabel = "workload.template.kubevirt.io/server"
+	TemplateFlavorLabel   = "flavor.template.kubevirt.io/medium"
+)
+
+// Operating Systems
+const (
+	DefaultWindows = "win10"
+	DefaultLinux   = "rhel8.1"
+	Unknown        = "unknown"
+)
+
+// Annotations
+const (
+	// CDI import backing file annotation on PVC
+	AnnImportBackingFile = "cdi.kubevirt.io/storage.import.backingFile"
+)
+
+// Regex which matches the snapshot identifier suffix of a
+// OVA disk backing file.
+var backingFilePattern = regexp.MustCompile("-\\d\\d\\d\\d\\d\\d.vmdk")
+
+// vSphere builder.
+type Builder struct {
+	*plancontext.Context
+	// MAC addresses already in use on the destination cluster. k=mac, v=vmName
+	macConflictsMap map[string]string
+}
+
+// Get list of destination VMs with mac addresses that would
+// conflict with this VM, if any exist.
+func (r *Builder) macConflicts(vm *model.VM) (conflictingVMs []string, err error) {
+	if r.macConflictsMap == nil {
+		list := []ocp.VM{}
+		err = r.Destination.Inventory.List(&list, base.Param{
+			Key:   base.DetailParam,
+			Value: "all",
+		})
+		if err != nil {
+			return
+		}
+
+		r.macConflictsMap = make(map[string]string)
+		for _, kVM := range list {
+			for _, iface := range kVM.Object.Spec.Template.Spec.Domain.Devices.Interfaces {
+				r.macConflictsMap[iface.MacAddress] = path.Join(kVM.Namespace, kVM.Name)
+			}
+		}
+	}
+
+	for _, nic := range vm.NICs {
+		if conflictingVm, found := r.macConflictsMap[nic.MAC]; found {
+			for i := range conflictingVMs {
+				// ignore duplicates
+				if conflictingVMs[i] == conflictingVm {
+					continue
+				}
+			}
+			conflictingVMs = append(conflictingVMs, conflictingVm)
+		}
+	}
+
+	return
+}
+
+// Create DataVolume certificate configmap.
+// No-op for OVA.
+func (r *Builder) ConfigMap(_ ref.Ref, _ *core.Secret, _ *core.ConfigMap) (err error) {
+	return
+}
+
+func (r *Builder) PodEnvironment(vmRef ref.Ref, sourceSecret *core.Secret) (env []core.EnvVar, err error) {
+	vm := &model.VM{}
+	err = r.Source.Inventory.Find(vm, vmRef)
+	if err != nil {
+		err = liberr.Wrap(
+			err,
+			"VM lookup failed.",
+			"vm",
+			vmRef.String())
+		return
+	}
+
+	env = append(
+		env,
+		core.EnvVar{
+			Name:  "V2V_vmName",
+			Value: vm.Name,
+		},
+	)
+
+	if strings.HasSuffix(vm.OvaPath, ".ova") {
+		env = append(env,
+			core.EnvVar{
+				Name:  "V2V_diskPath",
+				Value: vm.OvaPath,
+			},
+			core.EnvVar{
+				Name:  "V2V_provider",
+				Value: "ova",
+			})
+	} else {
```

Comment:
I think it should still be with `-i ova` and pointing to the folder, did you check this? (CC @nyoxi )

Reply:
it's just provided to the virt-v2v pod. based on that it uses later the `-i ova` in the command.